runtime/runtime.go

```diff
@@ -44,12 +44,12 @@ func New(opts *Options, logger *zap.Logger) (*Runtime, error) {
 	}
 
 	return &Runtime{
-		opts:         opts,
-		metastore:    metastore,
-		logger:       logger,
-		connCache:    newConnectionCache(opts.ConnectionCacheSize, logger),
-		catalogCache: newCatalogCache(),
-		queryCache:   newQueryCache(opts.QueryCacheSize),
+		opts:               opts,
+		metastore:          metastore,
+		logger:             logger,
+		connCache:          newConnectionCache(opts.ConnectionCacheSize, logger),
+		migrationMetaCache: newMigrationMetaCache(opts.ConnectionCacheSize), // todo :: different size for this ?
```

Comment:
If the migration meta gets evicted while it's used by a service somewhere, those mutations will be lost, right? So probably this needs to have infinite size until addressed by new controller?

Reply:
Yeah this makes sense. I will keep the size to max value.